tests/unit/repo/test_package_repository.py

```diff
@@ -64,160 +64,222 @@ def apt_repository():
+        create_apt(
             key_id="A" * 40,
             suites=["xenial", "xenial-updates"],
             url="http://archive.ubuntu.com/ubuntu",
         )
 
-    err = raised.value
-    assert str(err) == (
+    expected_message = (
         "Invalid package repository for 'http://archive.ubuntu.com/ubuntu': "
-        "no components specified."
-    )
-    assert err.details == "Components are required when using suites."
-    assert err.resolution == (
-        "Verify the repository configuration and ensure that 'components' "
-        "is correctly specified."
+        "components must be specified when using suites."
     )
 
+    err = raised.value
+    assert expected_message in str(err)
+
 
 def test_apt_invalid_missing_suites():
-    with pytest.raises(errors.PackageRepositoryValidationError) as raised:
-        PackageRepositoryApt(
+    with pytest.raises(pydantic.ValidationError) as raised:
+        create_apt(
             key_id="A" * 40,
             components=["main"],
             url="http://archive.ubuntu.com/ubuntu",
         )
 
-    err = raised.value
-    assert str(err) == (
+    expected_message = (
         "Invalid package repository for 'http://archive.ubuntu.com/ubuntu': "
-        "no suites specified."
-    )
-    assert err.details == "Suites are required when using components."
-    assert err.resolution == (
-        "Verify the repository configuration and ensure that 'suites' "
-        "is correctly specified."
+        "suites must be specified when using components."
     )
 
+    err = raised.value
+    assert expected_message in str(err)
+
 
 def test_apt_invalid_suites_as_path():
-    with pytest.raises(errors.PackageRepositoryValidationError) as raised:
-        PackageRepositoryApt(
+    with pytest.raises(pydantic.ValidationError) as raised:
+        create_apt(
             key_id="A" * 40,
             suites=["my-suite/"],
             url="http://archive.ubuntu.com/ubuntu",
         )
 
-    err = raised.value
-    assert str(err) == (
+    expected_message = (
         "Invalid package repository for 'http://archive.ubuntu.com/ubuntu': "
-        "invalid suite 'my-suite/'."
-    )
-    assert err.details == "Suites must not end with a '/'."
-    assert err.resolution == (
-        "Verify the repository configuration and remove the trailing '/' "
-        "from suites or use the 'path' property to define a path."
+        "invalid suite 'my-suite/'. Suites must not end with a '/'."
     )
 
+    err = raised.value
+    assert expected_message in str(err)
+
+
+@pytest.mark.parametrize(
+    "key_id,error",
+    [
+        ("ABCDE12345" * 4, None),
+        ("KEYID12345" * 4, "string does not match regex"),
+        ("abcde12345" * 4, "string does not match regex"),
+    ],
+)
+def test_apt_key_id(key_id, error):
+    repo = {
+        "type": "apt",
+        "url": "https://some/url",
+        "key-id": key_id,
+    }
+
+    if not error:
```

Comment:
It's probably best to break this into two tests — one that tests success and one that tests failures

Reply:
You are right of course! Updated